interview-questions.md

```diff
@@ -8,44 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a hash in Ruby?
 
-Your answer:
+Your answer: A hash is simply an object in ruby that contains only key value pairs.
 
-Researched answer:
+Researched answer: My answer was correct, missed a key detail that ruby hashes are unique in that the order of the keys don't matter.
 
 2. What is a gem?
 
-Your answer:
+Your answer: A gem is a packet of ruby code put together like a javascript package
 
-Researched answer:
+Researched answer: Yea, gems are just a Ruby specific way to talk about packages and instead of Yarn or Node, it uses RubyGems as the package manager.
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Ruby on Rails is an application that automates many of the things that need to be done to create a backend app.
 
-Researched answer:
+Researched answer: Ruby on Rails is actually an entire framework that handles a lot of the things done on the backend such as creating models, changing the shapes of databases, and queries.
```

Comment:
It is not just a backend app but a full stack application. This kind of difference is extremely important.